quinn-proto/src/connection/mod.rs

```diff
@@ -773,6 +775,46 @@ impl Connection {
 
         self.app_limited = buf.is_empty() && !congestion_blocked;
 
+        // Send MTU probe if necessary
+        if buf.is_empty() && self.state.is_established() {
+            let space_id = SpaceId::Data;
+            let probe_size = match self
+                .path
+                .mtud
+                .poll_transmit(self.spaces[space_id].next_packet_number)
+            {
+                Some(next_probe_size) => next_probe_size,
+                None => return None,
+            };
+
+            if buf.capacity() < probe_size as usize {
+                buf.reserve(probe_size as usize - buf.capacity());
+            }
```

Comment:
`reserve` is relative to the current `len`, not the capacity; further, we already know `len` is zero, and `reserve` already noops if redundant, so we might as well call it unconditionally.
```suggestion
            buf.reserve(probe_size as usize);
```

Reply:
Sure, I was a bit confused by the idea of tracking capacity separately from the buffer itself. I might explore extracting that logic into its own struct in a future PR, to remove that bit of complexity